File: smart-poultry-dashboard/src/types/enums.ts
export enum SensorType {
  TEMPERATURE = "temperature",
  HUMIDITY = "humidity",
  AMMONIA = "ammonia",
  LIGHT = "light",
  MOTION = "motion",
}

export enum DeviceType {
  FEEDER = "feeder",
  FAN = "fan",
  LIGHT = "light",
  HEATER = "heater",
}

export enum ActionValue {
  ON = "on",
  OFF = "off",
}

export enum Operator {
  GT = ">",
  GTE = ">=",
  LT = "<",
  LTE = "<=",
  EQ = "==",
}

export enum CommandStatus {
  PENDING = "pending",
  EXECUTED = "executed",
  FAILED = "failed",
}

export enum UserRole {
  ADMIN = "admin",
  MANAGER = "manager",
  WORKER = "worker",
}

File: app/models/enums.py
from enum import Enum

class SensorTypeEnum(str, Enum):
    TEMPERATURE = "temperature"
    HUMIDITY = "humidity"
    LIGHT = "light"
    WATER = "water"
    FEED = "feed"

class DeviceTypeEnum(str, Enum):
    FAN = "fan"
    FEEDER = "feeder"
    LIGHT = "light"
    ALARM = "alarm"

class OperatorEnum(str, Enum):
    GT = ">"
    LT = "<"
    GTE = ">="
    LTE = "<="
    EQ = "=="
    NEQ = "!="

class UserRoleEnum(str, Enum):
    ADMIN = "admin"
    OPERATOR = "operator"

class ActionValueEnum(str, Enum):
    ON = "ON"
    OFF = "OFF"

class CommandStatusEnum(str, Enum):
    pending = "pending"
    sent = "sent"
    failed = "failed"
    acknowledged = "acknowledged"